signature/cose/envelope.go

```diff
@@ -142,7 +150,10 @@ func (signer *localSigner) CertificateChain() []*x509.Certificate {
 }
 
 type envelope struct {
-	base *cose.Sign1Message
+	isSign      bool
```

Comment:
> Why do we need a flag to indicate if its a sign or verify call? Can we use the fact that sign and veify are different methods?

Code updated. But we still need to check if it is a sign or verify. This is because in base.Envelope.Sign(), we are also calling e.Envelope.Content(). However, due to Encode/Marshal, cose.Headers.Protected[SigningTime] is a cbor.RawMessage. Therefore, we need to Decode it back to time.Time during Sign. 
We don't need the Decode in verify though. Because when the upstream caller uses ParseEnvelope, Unmarshal is called (https://github.com/notaryproject/notation-core-go/blob/cd56ef60fb85f67d17bc59ff96637a93395c99a1/signature/cose/envelope.go#L158). go-cose will do the Decode for us.